src/client/mod.rs

```diff
@@ -97,6 +100,61 @@ impl GaloyClient {
         Ok(me)
     }
 
+    pub fn fetch_balance(
+        &self,
+        wallet: Option<Wallet>,
+        wallet_ids: Vec<String>,
+    ) -> anyhow::Result<String> {
+        let me = self.me()?;
+
+        let default_wallet_id = me.default_account.default_wallet_id;
+        let wallets = &me.default_account.wallets;
+
+        let wallet_ids_set: std::collections::HashSet<_> = wallet_ids.iter().collect();
+
+        let mut wallet_balances = String::new();
+
+        for wallet_info in wallets {
+            if wallet_ids_set.contains(&wallet_info.id)
+                || match (&wallet, &wallet_info.wallet_currency) {
+                    (Some(w), wc) if wallet_to_currency(w) == *wc => true,
+                    (None, _) if wallet_ids_set.is_empty() => true,
+                    _ => false,
+                }
+            {
+                let currency = match wallet_info.wallet_currency {
+                    WalletCurrency::USD => "cents",
+                    WalletCurrency::BTC => "sats",
+                    _ => "unknown",
+                };
+
+                wallet_balances += &format!(
+                    "{}{}: {} {}\n",
+                    if wallet_info.wallet_currency == WalletCurrency::USD {
+                        "USD wallet"
+                    } else if wallet_info.wallet_currency == WalletCurrency::BTC {
+                        "BTC wallet"
+                    } else {
+                        &wallet_info.id
+                    },
+                    if wallet_info.id == default_wallet_id {
+                        " (default)"
+                    } else {
+                        ""
+                    },
+                    wallet_info.balance,
+                    currency,
+                );
+            }
```

Comment:
This is very convoluted, can you simplify this?
Maybe with more Into impls?
Or a different struct with Display trait.

Reply:
Simplified the function logic and refactored the code by returning json, instead of string, as @nicolasburtey suggested [here](https://github.com/GaloyMoney/galoy-cli/pull/152#discussion_r1225741053).
b9a2227aabddad86e3691ec389637689c8d1ffa2

**New Sample Output:**

```
[
  {
    "currency": "Btc",
    "balance": "277401",
    "id": null,
    "default": true
  },
  {
    "currency": "Usd",
    "balance": "4892",
    "id": null,
    "default": false
  }
]
```